apps/server/src/modules/class/repo/classes.repo.ts

```diff
@@ -86,4 +57,13 @@ export class ClassesRepo {
 
 		return domainObject;
 	}
+
+	public async removeUserReference(userId: EntityId): Promise<number> {
+		const id = new ObjectId(userId);
+		const count = await this.em.nativeUpdate(ClassEntity, { $or: [{ userIds: id }, { teacherIds: id }] }, {
+			$pull: { userIds: id, teacherIds: id },
```

Comment:
so we handle it this way because we are sure that the same id would not be present in `userIds` as well as `teacherIds`? Not sure if it could happen, just asking to make sure.

Reply:
we try to remove that user id regardless if is student or teacher